testutil/containschecker.go

```diff
@@ -150,3 +150,69 @@ func (c *deepContainsChecker) Check(params []interface{}, names []string) (resul
 		return false, fmt.Sprintf("%T is not a supported container", container)
 	}
 }
+
+type deepContainerMatchChecker struct {
+	*check.CheckerInfo
+}
+
+// DeepContainerMatches checks if two containers contain the same elements in
+// the same number (but possibly different order) using DeepEqual. The container
+// can be an array, a slice or a map.
+var DeepContainerMatches check.Checker = &deepContainerMatchChecker{
+	&check.CheckerInfo{Name: "DeepContainerMatchChecker", Params: []string{"container1", "container2"}},
```

Comment:
I'm not sure about this name: the word "deep" already tells us that this works on containers, so the "container" part seems unnecessary. On the other hand, the most important characteristic of this matcher (the fact that it doesn't care about sorting) is not mentioned in its name. Maybe "DeepUnsortedMatchChecker" could be a better name?

Reply:
:+1: I went back and forth on the name but DeepUnsortedMatches and deepUnsortedMatchChecker seems clearer. I'll change it, thanks

Reply:
Thanks!